js-packages/@prestojs/rest/src/LimitOffsetPaginator.ts

```diff
@@ -0,0 +1,131 @@
+import { EndpointExecuteOptions } from './Endpoint';
+import Paginator, { PaginationStateChangeListener } from './Paginator';
+
+export type LimitOffsetPaginationState = {
+    limit?: number;
+    offset?: number;
+};
+
+export default class LimitOffsetPaginator extends Paginator {
+    total: number | null = null;
+
+    get limit(): number | null | undefined {
+        return this.currentState.limit;
+    }
+
+    get offset(): number {
+        return this.currentState.offset || 0;
+    }
+
+    constructor(
+        initialState: LimitOffsetPaginationState = {},
+        onChange?: PaginationStateChangeListener
+    ) {
+        super(initialState || {}, onChange);
+    }
+
+    setLimit(limit: number): void {
+        if (limit < 1) {
+            throw new Error(`Invalid limit ${limit} - should be >= 1`);
+        }
+        this.setState(currentState => {
+            // When page size changes we need to reset the offset
+            if (currentState.limit !== limit && currentState.offset) {
+                const nextState: LimitOffsetPaginationState = { ...currentState, limit };
+                delete nextState.offset;
+                return nextState;
+            }
+            return { ...currentState, limit };
+        });
+    }
+
+    setOffset(offset: number): void {
+        if (offset < 0) {
+            throw new Error(`Invalid offset ${offset} - should be >= 0`);
+        }
+        if (this.total != null && offset >= this.total) {
+            throw new Error(
+                `Invalid offset ${offset} - should be less than the total number of results (${this.total})`
+            );
```

Comment:
I've removed the restriction

There's no unnecessary restrictions in `next` and `prev` though? It just checks limit is known otherwise it's not possible to go next or prev